src/main/kotlin/HttpResponseBuilder.kt

```diff
@@ -0,0 +1,50 @@
+package response
+import httpstatus.HttpStatus
+
+class HttpResponseBuilder : ResponseBuilder {
+    private val protocol = "HTTP/1.1"
+    private val crlf = "\r\n"
+    private var httpStatusCode = HttpStatus.OK
+    private var httpBody = ""
+    private var httpHeaders: Map<String, String> = mapOf()
+
+    override fun build(): String {
+        val statusLine = generateStatusLine()
+        val responseHeaders = generateHeaders()
+        val responseBody = if (httpBody.isNotEmpty()) setContentLength() + "$crlf$httpBody" else ""
+        reset()
+        return statusLine + responseHeaders + responseBody
+    }
+
+    override fun setStatusCode(statusCode: HttpStatus) {
+        httpStatusCode = statusCode
+    }
+
+    override fun setBody(body: String) {
+        httpBody = body
+    }
+
+    override fun setHeaders(headers: Map<String, String>) {
+        httpHeaders = headers
+    }
+
+    override fun reset() {
```

Comment:
instead of calling reset, could you just pass in a new instance of the class?

Reply:
Do you mean to pass a new instance of `ResponseBuilder` in the `Action`'s `act(responseBuilder: ResponseBuilder)`?

Reply:
Yep, that's what I was thinking. I see that it might be harder than I thought.

In `server.kt`, I thought you could pass in a new instance here.
```
val response = action.act(responseBuilder)
```

I think the reason you can't just pass in a new instance is because ResponseBuilder is passed in as an argument into `Server`, right?

```
class Server(private val serverSocket: ServerSocket,
             private val parser: RequestParser,
             private val responseBuilder: ResponseBuilder,
             private val router: Router) {
```

Not exactly sure what the best option is. You could pass in a `ResponseBuilderGenerator` that returns a `ResponseBuilder` into `Server`.

Reply:
Passing a `ResponseBuilderGenerator` is very possible, and could be used in multiple ways to achieve what we're talking about, but I'm curious to know if it is necessary? The motivation behind the `reset()` method came from the [Builder Pattern](https://refactoring.guru/design-patterns/builder). In the article a `reset()` method is used so the builder can be reused to produce another product after delivering the result. In the context of this project, I use `reset()` to reset the varying states of the `HttpResponseBuilder`; `httpStatusCode`, `httpBody`, and `httpHeaders` so I can reuse the class to build a response.